### PR TITLE
refactor: use fully-qualified paths in Compact derives

### PR DESCRIPTION
**Pull request by 0xDeon**
This PR addresses issue #11862 by removing feature-gated imports of reth_codecs::Compact through the use of fully qualified paths in generated code. 

Additionally, a crate attribute (#[reth_codecs(crate = "...")]) was added, allowing the use of custom paths for reth_codecs in generated implementations."

Changes Made:
     - Updated generator.rs to use reth_codecs::Compact directly.
     - Added parsing for the crate attribute to specify alternate paths for reth_codecs.
     - Refactored quote! blocks to support the new attribute, defaulting to reth_codecs if none is provided.

Testing:
     - All tests ran successfully.

### CODE REVIEW

File: crates/primitives/src/receipt.rs [hunk 1]
```diff
@@ -128,11 +127,12 @@ impl From<Receipt> for ReceiptWithBloom {
     }
 }
 
-/// [`Receipt`] with calculated bloom filter.
+/// [Receipt] with calculated bloom filter.
 #[derive(Clone, Debug, PartialEq, Eq, Default, Serialize, Deserialize)]
 #[cfg_attr(any(test, feature = "arbitrary"), derive(arbitrary::Arbitrary))]
 #[cfg_attr(any(test, feature = "reth-codec"), derive(reth_codecs::Compact))]
 #[cfg_attr(any(test, feature = "reth-codec"), reth_codecs::add_arbitrary_tests(compact))]
+#[reth_codecs(crate = "reth_codecs")]
```
Comment (mattsse) [inline crates/primitives/src/receipt.rs:135]: this shouldn't require the attribute?

File: crates/primitives/src/receipt.rs [hunk 2]
```diff
@@ -205,6 +205,16 @@ impl<'a> arbitrary::Arbitrary<'a> for Receipt {
...
+impl ReceiptWithBloom {
+    /// Returns the enveloped encoded receipt.
+    ///
+    /// See also [ReceiptWithBloom::encode_enveloped]
+    pub fn envelope_encoded(&self) -> Bytes {
+        let mut buf = Vec::new();
+        self.encode_enveloped(&mut buf);
+        buf.into()
+  }
+}
```
Comment (mattsse) [inline crates/primitives/src/receipt.rs:217]: this is unrelated?

File: crates/primitives/src/receipt.rs [hunk 3]
```diff
@@ -454,7 +464,7 @@ impl ReceiptWithBloomEncoder<'_> {
     fn encode_inner(&self, out: &mut dyn BufMut, with_header: bool) {
         if matches!(self.receipt.tx_type, TxType::Legacy) {
             self.encode_fields(out);
-            return
+            return;
```
Comment (mattsse) [inline crates/primitives/src/receipt.rs:467]: please ser, no more additional `;`, I can't take this anymore :D

File: crates/primitives/src/receipt.rs [hunk 4]
```diff
@@ -11,23 +11,22 @@ use alloy_rlp::{length_of_length, Decodable, Encodable, RlpDecodable, RlpEncodab
 use bytes::{Buf, BufMut};
 use core::{cmp::Ordering, ops::Deref};
 use derive_more::{DerefMut, From, IntoIterator};
-#[cfg(feature = "reth-codec")]
-use reth_codecs::Compact;
```
Comment (mattsse) [inline crates/primitives/src/receipt.rs:15]: cool

File: crates/primitives/src/receipt.rs [hunk 5]
```diff
@@ -81,22 +80,22 @@ impl Receipt {
     IntoIterator,
 )]
 pub struct Receipts {
-    /// A two-dimensional vector of optional `Receipt` instances.
+    /// A two-dimensional vector of optional Receipt instances.
```
Comment (mattsse) [inline crates/primitives/src/receipt.rs:83]: please keep this, optionally this should be `[`Receipt`]` instead.

File: crates/primitives/src/receipt.rs [hunk 6]
```diff
@@ -47,19 +46,19 @@ pub struct Receipt {
...
 
-    /// Calculates the bloom filter for the receipt and returns the [`ReceiptWithBloom`] container
+    /// Calculates the bloom filter for the receipt and returns the [ReceiptWithBloom] container
     /// type.
     pub fn with_bloom(self) -> ReceiptWithBloom {
         self.into()
     }
 
-    /// Calculates the bloom filter for the receipt and returns the [`ReceiptWithBloomRef`]
+    /// Calculates the bloom filter for the receipt and returns the [ReceiptWithBloomRef]
```
Comment (mattsse) [inline crates/primitives/src/receipt.rs:61]: we mostly settled on the `[``]` variant, so please undo these

File: crates/storage/codecs/derive/src/compact/generator.rs [hunk 7]
```diff
@@ -232,3 +236,26 @@ fn generate_to_compact(fields: &FieldList, ident: &Ident, is_zstd: bool) -> Vec<
 
     lines
 }
+
+/// Function to extract the crate path from #[reth_codecs(crate = "...")] attribute.
+
+fn parse_reth_codecs_path(attrs: &[Attribute]) -> TokenStream2 {
```
Comment (mattsse) [inline crates/storage/codecs/derive/src/compact/generator.rs:242]: if there's no `crate` attribute then this should fallback to  just `reth_codecs`

**Review by mattsse — CHANGES_REQUESTED** [review]
this looks great, ty.

I only have some nits re additional `;` or unrelated code changes

**Comment by DanielEmmanuel1** [comment]
@mattsse Can i get a 24 hours extension
I'll resolve all the issues by then

Is that okay?

**Comment by mattsse** [comment]
of course, there's no deadline ;) so please take your time

appreciate your contribution 🫡

File: crates/primitives/src/receipt.rs (hunk 1 above)
Comment (DanielEmmanuel1) [reply crates/primitives/src/receipt.rs:135]: I didn't see the need for it here

**Comment by DanielEmmanuel1** [comment]
@mattsse I seem to have hit a brick wall, i have handled most of the issues above but for some reason i can't seems to get the default path the reth_codecs in the generator, can u maybe clarify how i could do this?

File: crates/storage/codecs/derive/src/compact/generator.rs [hunk 8]
```diff
@@ -1,11 +1,13 @@
-//! Code generator for the `Compact` trait.
+//! Code generator for the Compact trait.
```
Comment (onbjerg) [inline crates/storage/codecs/derive/src/compact/generator.rs:1]: please revert all of these doc changes, especially the ones that link to structs

Comment (DanielEmmanuel1) [reply crates/storage/codecs/derive/src/compact/generator.rs:1]: I have already taken care of most of the issues brought up by @mattsse , I just haven't pushed yet cause on the last issue, I need clarifications on that @onbjerg

**Comment by DanielEmmanuel1** [comment]
If no crate is provided, I understand the fallback should be reth_codecs. Since this trait is used in the reth_codecs crate too, without specifying a crate. It attempts to use the default reth_codecs this throws an error that the crate or module reth_codecs cannot be resolved. The issue is how to get the default path to work across crates and in the reth_codecs crate itself @mattsse

**Comment by DanielEmmanuel1** [comment]
@mattsse @onbjerg All done please review

**Review by mattsse — CHANGES_REQUESTED** [review]
changes lgtm,

please undo all unrelated changes to comments

**Comment by DanielEmmanuel1** [comment]
Okay, a minute

**Comment by DanielEmmanuel1** [comment]
@mattsse Please review, all linting issues sorted
